File: objc/Command/Command/GiftFriendCommand.m
#import "GiftFriendCommand.h"

@implementation GiftFriendCommand

-(instancetype)initWithBook:(Book*)book friendName:(NSString*)friendName {
    if(self = [super init]) {
        _book = book;
        _friendName = friendName;
    }
    return self;
}

-(void)execute {
    [_book giftToFriend:_friendName];
}

@end

File: swift/Command/Command/GiftFriendCommand.swift
import Foundation

public class GiftFriendCommand : Command {
    
    let book: Book
    let friendName: String
    
    init(_ book: Book, _ friendName: String) {
        self.book = book
        self.friendName = friendName
    }
    
    func execute() {
        book.giftToFriend(friendName)
    }
    
}
